### ISSUE TITLE
PostgreSQL: Support all comparison operators in parser

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#9722]
PostgreSQL defines a few non-standard comparison operators which jOOQ's parser should also support. See [documentation](https://www.postgresql.org/docs/current/functions-comparison.html) for details.

- [ ] expression ISNULL
- [ ] expression NOTNULL
- [ ] boolean_expression IS TRUE
- [ ] boolean_expression IS NOT TRUE
- [ ] boolean_expression IS FALSE
- [ ] boolean_expression IS NOT FALSE
- [ ] boolean_expression IS UNKNOWN
- [ ] boolean_expression IS NOT UNKNOWN

**Comment by lukaseder** [comment]
I'm not convinced we need to support `ISNULL` and `NOTNULL` from PostgreSQL. The other predicates are standard SQL and are covered in https://github.com/jOOQ/jOOQ/issues/4153

I'm closing this as a duplicate of #4153

**Comment by lukaseder** [comment]
I see, this was mostly meant as a parser improvement issue. So, I'll change the issue as follows:

- Remove the standard SQL `IS TRUE` etc. predicates, as I'd like to address those more thoroughly via #4153
- Make this about supporting the PostgreSQL `ISNULL` and `NOTNULL` unary operators

I still don't think we should address this soon, because such unary operators tend to lead to a ton of ambiguities (e.g. are they column aliases or operators?). But we can keep it open to collect user feedback.

**Comment by knutwannheden** [comment]
Makes sense. I didn't even know that `IS [ NOT ] { TRUE | FALSE }` is standard SQL.

**Comment by lukaseder** [comment]
> Makes sense. I didn't even know that `IS [ NOT ] { TRUE | FALSE }` is standard SQL.

Well, not the most useful predicates... The main benefit is that they are "null safe".
